xdsl/traits.py

```diff
@@ -498,3 +498,21 @@ def has_effects(cls, op: Operation) -> bool:
 
 class Pure(NoMemoryEffect):
     """A trait that signals that an operation has no side effects."""
+
+
+class HasInsnRepresentation(OpTrait, abc.ABC):
+    """
+    A trait providing information on how to encode an operation using a .insn assember directive.
+
+    The returned string contains python string.format placeholders where formatted operands are inserted during
+    printing.
+
+    See https://sourceware.org/binutils/docs/as/RISC_002dV_002dDirectives.html for more information.
+    """
+
+    @abc.abstractmethod
```

Comment:
What happens when you remove this? Since you provide an implementation, does it actually need the decorator?

Reply:
Pyright helpfully warns users if a method with this decorator has not been implemented

Reply:
With an `@abstractmethod` you only need to say `...` instead of a function body. Since the method throws a `NotImplementedError`, my suspicion is one, that we probably don't need both, and two, that it might potentially make the other error go away if we lose the decorator.